src/database-sqlite3.cpp

```diff
@@ -136,6 +137,23 @@ void Database_SQLite3::endSave() {
 	SQLRES(sqlite3_step(m_stmt_end), SQLITE_DONE,
 		"Failed to commit SQLite3 transaction");
 	sqlite3_reset(m_stmt_end);
+
+	int total_size;
+	int checkpointed;
+	SQLOK(sqlite3_wal_checkpoint_v2(m_database, NULL,
+			SQLITE_CHECKPOINT_PASSIVE, &total_size, &checkpointed),
+		"failed to checkpoint database");
+	if (total_size >= 0 && checkpointed >= 0) {
+		int new_wal_backlog = total_size - checkpointed;
+		// Normally, sqlite would checkpoint every 1000 pages.
+		if (m_wal_backlog / 10000 < new_wal_backlog / 10000)
```

Comment:
The code prints one message each time the WAL backlog has crossed the next multiple of 10000 frames. Each subsequent crossing will usually happen after a number of checkpoints, each adding (much) less than 10000 to the backlog. So, the difference between the previous and the current backlog will rarely if ever exceed 10000.

The purpose is to avoid flooding the log with messages when the backlog has grown large (e.g. one at every checkpoint), while still regularly printing a message if it keeps growing.

The code obviouly needs a clarifying comment.